virttest/utils_misc.py

```diff
@@ -4024,26 +4025,35 @@ def get_model_features(model_name):
     """
     features = []
     conf = "/usr/share/libvirt/cpu_map.xml"
+    conf_dir = "/usr/share/libvirt/cpu_map/"
 
     try:
-        output = open(conf, 'r').read()
-        root = ET.fromstring(output)
-        # Find model
-        for model_n in root.findall('arch/model'):
-            if model_n.get('name') == model_name:
-                model_node = model_n
-                for feature in model_n.findall('feature'):
-                    features.append(feature.get('name'))
-                break
-        # Handle nested model
-        nested_model = model_node.find('model')
-        if nested_model is not None:
-            nested_model_name = nested_model.get('name')
-            for model_n in root.findall('arch/model'):
-                if model_n.get('name') == nested_model_name:
-                    for feature in model_n.findall('feature'):
-                        features.append(feature.get('name'))
-                    break
+        if os.path.exists(conf):
+            with open(conf, 'r').read() as output:
+                root = ET.fromstring(output)
+                while True:
+                    # Find model in file /usr/share/libvirt/cpu_map.xml
+                    for model_n in root.findall('arch/model'):
+                        if model_n.get('name') == model_name:
+                            model_node = model_n
+                            for feature in model_n.findall('feature'):
+                                features.append(feature.get('name'))
+                            break
+                    # Handle nested model
+                    if model_node.find('model') is not None:
+                        model_name = model_node.find('model').get('name')
+                        continue
+                    else:
+                        break
+            # Find model in dir /usr/share/libvirt/cpu_map
+        if os.path.exists(conf_dir):
+            filelist = os.listdir(conf_dir)
+            for file_name in filelist:
+                if model_name in file_name:
+                    with open(conf_dir + file, "r").read() as output:
```

Comment:
1. It would break here as `file_name` to be used ?
2. please use `os.path.join()`